File: examples/js/blockchaincom-private-api.js
"use strict";

const ccxt = require('../../ccxt.js')
const asTable = require('as-table')
const log = require('ololog').configure({ locate: false })

require('ansicolor').nice

    ; (async () => {

        let exchange = new ccxt.blockchaincom({
            'secret': 'YOUR_SECRET_KEY'
        })
        let withdrawal_beneficiary = 'BENEFICIARY'

        try {

            // fetch fees
            let tradingFees = await exchange.fetchTradingFees()
            log('tradingFees'.green, tradingFees)

            // my trades
            let myTrades = await exchange.fetchMyTrades('BTC-USD', undefined, 5)
            log('myTrades'.green, asTable(myTrades))

            // deposits
            let deposits = await exchange.fetchDeposits()
            log('Deposits'.green, asTable(deposits))

            // fetch account balance from the exchange
            let balance = await exchange.fetchBalance()
            log('balance'.green, balance.total)

             // withdrawals
            let withdrawals = await exchange.fetchWithdrawals()
            log('Withdrawals'.green, asTable(withdrawals))

            // deposit address
            let paxDeposit = await exchange.fetchDepositAddress("XLM")
            log('Pax deposit address'.green, paxDeposit)

            // create new limit order
            let newOrder = await exchange.createOrder('BTC/USDT', 'limit', 'buy', 0.01, 2000)
            console.log('New limit order'.green, newOrder);

            // open orders
            let openOrders = await exchange.fetchOpenOrders()
            log('Open orders'.green, asTable(openOrders))

            // cancel order
            let canceledOrder = await exchange.cancelOrder(newOrder.id)
            console.log('Canceled order'.green, canceledOrder);

            // closed orders
            let closedOrders = await exchange.fetchClosedOrders()
            log('Closed orders'.green, asTable(closedOrders))

            // orders by state
            let ordersByState = await exchange.fetchOrdersByState("CANCELED")
            log('Orders by State'.green, asTable(ordersByState))

            // fetch withdrawal white list
            let whiteList = await exchange.fetchWithdrawalWhitelist()
            log('whiteList'.green, asTable(whiteList))

            // cancel all open orders
            let canceledOrders = await exchange.cancelOrders(null)
            console.log('Canceled all open orders'.green, canceledOrders);

            // withdrawal
            const params = { 'beneficiary': withdrawal_beneficiary }
            let btcWithdraw = await exchange.withdraw("BTC", 0.01, "", undefined, params)
            log('BTC withdrawal'.green, btcWithdraw)

        } catch (e) {

            if (e instanceof ccxt.DDoSProtection || e.message.includes('ECONNRESET')) {
                log.bright.yellow('[DDoS Protection] ' + e.message)
            } else if (e instanceof ccxt.RequestTimeout) {
                log.bright.yellow('[Request Timeout] ' + e.message)
            } else if (e instanceof ccxt.AuthenticationError) {
                log.bright.yellow('[Authentication Error] ' + e.message)
            } else if (e instanceof ccxt.ExchangeNotAvailable) {
                log.bright.yellow('[Exchange Not Available Error] ' + e.message)
            } else if (e instanceof ccxt.ExchangeError) {
                log.bright.yellow('[Exchange Error] ' + e.message)
            } else if (e instanceof ccxt.NetworkError) {
                log.bright.yellow('[Network Error] ' + e.message)
            } else {
                throw e;
            }
        }

    })()

File: examples/py/blockchaincom-private-api.py
import ccxt

exchange = ccxt.blockchaincom({
    'secret': 'YOUR_SECRET_KEY',
    'timeout': 30000,
})

symbol = 'BTC/USDT'
type = 'limit' #or 'market', or 'Stop' or 'StopLimit'
side = 'buy'
amount = 0.01
price = 2000

withdrawal_beneficiary = 'BENEFICIARY'

fees = exchange.fetch_trading_fees()
print(fees)

print("\n")

mytrades = exchange.fetch_my_trades(symbol)
print(mytrades)

print("\n")

deposits = exchange.fetch_deposits()
print(deposits)

print("\n")

balance = exchange.fetch_balance()
print(balance)

print("\n")

withdrawals = exchange.fetch_withdrawals()
print(withdrawals)

print("\n")

address = exchange.fetch_deposit_address("XLM")
print(address)

print("\n")

new_order = exchange.create_order(symbol, type, side, amount, price)
print(new_order)

print("\n")

open_orders = exchange.fetch_open_orders()
print(open_orders)

print("\n")

cancel_order = exchange.cancel_order(new_order['id'])
print(cancel_order)

print("\n")

closed_orders = exchange.fetch_closed_orders()
print(closed_orders)

print("\n")

orders_by_state = exchange.fetch_orders_by_state("CANCELED")
print(orders_by_state)

print("\n")

white_list = exchange.fetch_withdrawal_whitelist()
print(white_list)

print("\n")

white_list_by_currency = exchange.fetch_withdrawal_whitelist_by_currency('BTC')
print(white_list_by_currency)

print("\n")

canceledOrders = exchange.cancel_orders(None)
print(canceledOrders)

print("\n")

params = { 'beneficiary': withdrawal_beneficiary }
btcWithdraw = exchange.withdraw("BTC", amount, None, None, params)
print(btcWithdraw)